demo/test/src/components/DropdownMultiselect.js

```diff
@@ -0,0 +1,235 @@
+import React, {useEffect, useRef, useState} from "react";
+import PropTypes from "prop-types";
+
+const DropdownMultiselect = (props) => {
+  
+  const [showDropdown, setShowDropdown] = useState(false);
+  const [options, setOptions]           = useState([]);
+  const [selected, setSelected]         = useState([]);
+  const node                            = useRef(null);
+
+  const handleClickOutside = (ev) => {
+    if (showDropdown !== false &&
+      node.current.target == ev.target
+    ) {
+      setShowDropdown(false);
+    }
+  }
+  const handleClick = () => {
+    setShowDropdown(!showDropdown);
+  }
+  const handleChange = (ev) =>  {
+    if(ev.currentTarget === undefined) {
+      return;
+    }
+    
+    var currentSelected = [...selected];
+    let value = ev.currentTarget.value;
+    
+    if (ev.currentTarget.checked) {
+      currentSelected.push(value);
+    } else {
+      var index = currentSelected.indexOf(value);
+      currentSelected.splice(index, 1);
+    }
+
+    // update the state with the new array of options
+    setSelected(currentSelected);
+   
+    if(props.handleOnChange !== undefined) {
+      props.handleOnChange(currentSelected)
+    }
+  }
+  const getPlaceholderValue = () => {
+    if (selected.length == 0) {
+      return props.placeholder;
+    }
+
+    if (
+      props.placeholderMultipleChecked !== null &&
+      selected.length > 1
+    ) {
+      return props.placeholderMultipleChecked;
+    } else {
+      let currentOptions = [...options];
+
+      if(currentOptions.length == 0) {
+        return props.placeholder;
+      }
+
+      let selectedLabels = [];
+      let optionKey = props.optionKey;
+      
+      selected.map((row) => {
+        let foundOption = currentOptions.find((option) => {
+          return option['key'] == row;
+        });
+
+        selectedLabels.push(foundOption.label);
+      });
+
+      return selectedLabels.join(", ");
+    }
+  }
+  const handleSelectDeselectAll = () => {
+    if (selected.length == options.length) {
+      setSelected([]);
+      if(props.handleOnChange) {
+        props.handleOnChange([]);
+      }
+    } else {
+      let allOptions = options;
+
+      let newSelected = [];
+
+      allOptions.map((obj) => {
+        newSelected.push(obj.key.toString());
+      });
+
+      setSelected(newSelected);
+      if(props.handleChange) {
+        props.handleOnChange(newSelected)
+      }
```

Comment:
also right here the select all was not firing the call back for whatever reason, it looked like your code should have worked but I added this, not sure if its the best way. After the previous if block.

      if(props.handleOnChange !== undefined) {
        props.handleOnChange(newSelected)
      }